lib/webpacker/dev_server.rb

```diff
@@ -0,0 +1,73 @@
+# Same convention as manifest/configuration.rb
+
+# Loads webpacker configuration from config/webpacker.yml
+
+require "webpacker/configuration"
+
+class Webpacker::DevServer < Webpacker::FileLoader
+  class << self
+    def enabled?
+      case ENV["WEBPACKER_DEV_SERVER"]
+      when /true/i then true
+      when /false/i then false
+      else !data[:dev_server].nil?
+      end
+    end
+
+    # read settings for dev_server
+    def hot_reloading?
+      if enabled?
+        case ENV["WEBPACKER_HMR"]
+        when /true/i then true
+        when /false/i then false
+        else fetch(:hot_reloading)
+        end
+      else
+        false
```

Comment:
Again, as above, I think we should refocus this. The dev-server is there when you want live-reloading (and potentially HMR). On-demand compilation, the new default, is there when . you just want "static files" (which, coincidentally, is where we ended up at Basecamp as well -- we didn't like live reloading either).